docs/cunumeric/source/user/notebooks/black_scholes.ipynb

```diff
@@ -0,0 +1,233 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 82,
+   "id": "597c89b9-6220-4360-82da-fb2aac8e70a5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cunumeric as np  # instead of numpy"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c8545c68-7ea0-4eeb-b290-e0643e2833b9",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "The Cumulative Normal Distribution (CND) is defined as \n",
+    "$$\n",
+    "\\frac{1}{2\\pi}\\int_{-\\infty}^{x} e^{-t^2/2}\\; dt = \\frac12\\big[1 + \\mathtt{erf}\\big(\\frac{x}{\\sqrt{2}}\\big)\\big]\n",
+    "$$\n",
+    "\n",
+    "In order to compute this efficiently, an expansion approximation (from John C. Hull (1997) “Options, Futures, and Other Derivatives”) is used."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 90,
+   "id": "b798400e-b726-46cf-83de-cbe0e613737f",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def CND(d):\n",
+    "    # The expansion coefficients for erf are derived from Abramowitz and Stegun, eq 7.1.26 \n",
+    "    # https://www.cs.bham.ac.uk/~aps/research/projects/as/resources/AandS-a4-v1-2.pdf\n",
+    "    A1 = 0.31938153\n",
+    "    A2 = -0.356563782\n",
+    "    A3 = 1.781477937\n",
+    "    A4 = -1.821255978\n",
+    "    A5 = 1.330274429\n",
+    "    RSQRT2PI = 0.39894228040143267793994605993438  # 1 / sqrt(2*pi)\n",
+    "\n",
+    "    K = 1.0 / (1.0 + 0.2316419 * np.abs(d))\n",
+    "\n",
+    "    cnd = RSQRT2PI * np.exp(-0.5*d*d) * (K*(A1 + K*(A2 + K*(A3 + K*(A4 + K*A5)))))\n",
+    "\n",
+    "    return np.where(d > 0, 1.0 - cnd, cnd)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8e4ae70e-30ac-47cf-949d-444695b5cacc",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Now that we have a method to compute the CND, we can imlement the Black-Scholes formula below"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 91,
+   "id": "fc601408-39db-4149-afd6-2866eab73f8f",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def black_scholes(S, X, T, r, v):\n",
+    "    sqrt_T = np.sqrt(T)\n",
+    "    \n",
+    "    d1 = np.log(S / X) + (r + 0.5 * v * v) * T / (v * sqrt_T)\n",
+    "    cnd_d1 = CND(d1)\n",
+    "\n",
+    "    d2 = d1 - v * sqrt_T\n",
+    "    cnd_d2 = CND(d2)\n",
+    "    \n",
+    "    exp_rT = np.exp(-r * T)\n",
+    "    \n",
+    "    call_result = S * cnd_d1 - X * exp_rT * cnd_d2\n",
+    "    put_result = X * exp_rT * (1.0 - cnd_d2) - S * (1.0 - cnd_d1)\n",
+    "\n",
+    "    return call_result, put_result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "31b2da3e-1c67-4f07-a485-77320ec0210e",
+   "metadata": {},
+   "source": [
+    "Let's use the `black_scholes` function above to price one put and call million options, keeping everying fixed except the current option price $S$. "
```

Comment:
should this be "one million put and call options"?

Reply:
https://github.com/nv-legate/cunumeric/pull/805/commits/971855c33f59ee06de0f249dd4a9640685234025